library/osrm.hpp

```diff
@@ -49,9 +49,9 @@ class OSRM
     std::unique_ptr<OSRM_impl> OSRM_pimpl_;
 
   public:
-    explicit OSRM(libosrm_config &lib_config);
+    explicit OSRM(LibOSRMConfig lib_config);
```

Comment:
If you take `LibOSRMConfig` by value here it can not be an [incomplete type](http://en.cppreference.com/w/cpp/language/type#Incomplete_type). If you instead take it by const-ref, you can simply forward-declare `LibOSRMConfig` above, and then remove the appropriate include, keeping this header clean by breaking dependencies.